src/e2e-test/java/io/cdap/plugin/servicenowsink/stepsdesign/DesignTimeSteps.java

```diff
@@ -16,26 +16,60 @@
 import org.apache.oltu.oauth2.common.exception.OAuthProblemException;
 import org.apache.oltu.oauth2.common.exception.OAuthSystemException;
+
+import java.io.IOException;
+
 /**
  * ServiceNow sink - Properties page - Steps.
  */
 
 public class DesignTimeSteps {
+  public static String query;
+
+  @And("Verify If new record created in ServiceNow application for table {string} is correct")
+  public void verifyIfNewRecordCreatedInServiceNowApplicationForTableIsCorrect(String tableName)
+    throws IOException, InterruptedException, OAuthProblemException, OAuthSystemException {
+    String tableValueFromPluginPropertiesFile = PluginPropertyUtils.pluginProp(tableName);
+
+    switch (tableValueFromPluginPropertiesFile) {
+      case "proc_rec_slip_item" :
+        query = "number=" + TestSetupHooks.receivingSlipLineRecordUniqueNumber;
+        ServiceNowSinkPropertiesPageActions.verifyIfRecordCreatedInServiceNowIsCorrect(query,
+                                                                            tableValueFromPluginPropertiesFile);
+        break;
+      case "agent_assist_recommendation":
+        query = "name=" + TestSetupHooks.agentAssistRecommendationUniqueName;
+        ServiceNowSinkPropertiesPageActions.verifyIfRecordCreatedInServiceNowIsCorrect(query,
+                                                                             tableValueFromPluginPropertiesFile);
+        break;
+      case "pc_vendor_cat_item":
+        query = "sys_update_name=" + TestSetupHooks.vendorCatalogItemUniqueName;
+        ServiceNowSinkPropertiesPageActions.verifyIfRecordCreatedInServiceNowIsCorrect(query,
+                                                                              tableValueFromPluginPropertiesFile);
+        break;
+      case "service_offering":
+        query = "number=" + TestSetupHooks.serviceOfferingUniqueNumber;
+        ServiceNowSinkPropertiesPageActions.verifyIfRecordCreatedInServiceNowIsCorrect(query,
+                                                                              tableValueFromPluginPropertiesFile);
+        break;
+      default:
+        throw new IllegalStateException("Unexpected value: " + tableValueFromPluginPropertiesFile);
+    }
+  }
 
-  @Then("Verify If new record is created in ServiceNow application")
-  public void verifyIfRecordExistsinServiceNowTable() throws OAuthProblemException, OAuthSystemException {
-    ServiceNowSinkPropertiesPageActions.verifyIfRecordExistsInServiecNowReceivingSlipLineTable(
-      TestSetupHooks.receivingSlipLineRecordUniqueNumber);
- }
+  @Then("Verify If an updated record in ServiceNow application for table {string} is correct")
+  public void verifyIfAnUpdatedRecordInServiceNowApplicationForTableIsCorrect(String tableName)
+    throws OAuthProblemException, OAuthSystemException, IOException, InterruptedException {
+    String tableValueFromPluginPropertiesFile = PluginPropertyUtils.pluginProp(tableName);
+    query = "sys_id=" + TestSetupHooks.systemId;
```

Comment:
use ServiceNowConstants variable for sys_id

Reply:
Done